### DISCUSSION TITLE
Blacklineing an image in PDF won't remove below content in version 1.22.2

### DISCUSSION THREAD

**Discussion by bbustamante** [pymupdf/PyMuPDF#2399] [Looking for help] [answered]
Hello,
I have just updated my PyMuPDF version from 1.21.1 to 1.22.2. When I blackline a text from an image even wen I edit the result PDF to get de blankline from the file manualy, the text behind it was desaparead (that's, for me, the correct working system). But right now, when I do it the text is still there. This just happend when the text is an image inside de PDF (second line of the example), not wen is a real text (first line of the example).

I include three (plus one) images with the process and the PDF I use:

1) Te original PDF with two lines. First one is real text while the second one is an image.
![image](https://github.com/pymupdf/PyMuPDF/assets/2870196/d291214b-476c-4ec5-9c1a-8477353c6a78)

2) The result PDF after blackileing it with PyMuPDF.
![image](https://github.com/pymupdf/PyMuPDF/assets/2870196/2a441c11-36bf-4af3-ae13-c43494d89822)

3) The result PDF manually edited to remove the blacklines.
![image](https://github.com/pymupdf/PyMuPDF/assets/2870196/3553329c-d760-46e5-ac91-e094b77fb426)

4) The PDF file I use for the example.
[PyMuPDF example.pdf](https://github.com/pymupdf/PyMuPDF/files/11443267/PyMuPDF.example.pdf)

5) The result of the example made with PyMuPDF 1.21.1 after editing manually for removing blacklines.
![image](https://github.com/pymupdf/PyMuPDF/assets/2870196/c73b17f7-d579-442f-bb39-40e474b94dcb)

Any idea of what is happening? How could I resolve it?
Thanks.

**Comment by JorjMcKie** [comment]
What is the code you are using?

**Reply by bbustamante** [reply, marked as answer]
```
#!/usr/bin/env python

import fitz

# Open the PDF file
doc = fitz.open("PyMuPDF example.pdf")

# Get the first page of the document
page = doc[0]

#### BEÑAT BLACKLINE TEXT ####

# Define the area to be redacted
rect = fitz.Rect(133.01550874920002, 87.82992395007999, 163.929577416, 94.73601638912)

# Create a redaction annotation with black color fill
page.add_redact_annot(rect, fill=(0, 0, 0))

#### BUSTAMANTE BLACKLINE TEXT ####

# Define the area to be redacted
rect = fitz.Rect(163.85502549240002, 87.91132582719999, 219.9595896564, 94.70768072959999)

# Create a redaction annotation with black color fill
page.add_redact_annot(rect, fill=(0, 0, 0))

#### BEÑAT BLACKLINE IMAGE ####

# Define the area to be redacted
rect = fitz.Rect(138.41771032320003, 165.5578356256, 169.47409023600002, 172.7324736032)

# Create a redaction annotation with black color fill
page.add_redact_annot(rect, fill=(0, 0, 0))

#### BUSTAMANTE BLACKLINE IMAGE ####

# Define the area to be redacted
rect = fitz.Rect(169.3404766152, 165.70278057279998, 226.11229924800003, 172.525437056)

# Create a redaction annotation with black color fill
page.add_redact_annot(rect, fill=(0, 0, 0))

# Apply the redactions to the page
page.apply_redactions()

# Save the modified PDF file
doc.save('PyMuPDF example blacklined.pdf')
```

**Reply by JorjMcKie** [reply]
This discussion refers to issue #2404 which has already been investigated and a fix has been identified.
It will be published in a future version.
